### PR TITLE
fix: fix submenu header double line

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This fixes issue where title of submenu is too long for column and would make text break to next line. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
before

Screen Shot 2021-05-13 at 6 24 06 PM
Screen Shot 2021-05-13 at 6 24 30 PM

after

https://user-images.githubusercontent.com/17326228/118207199-fd7d6380-b418-11eb-90a9-fe114fb7e073.mov

### TEST PLAN
Go to alert & reports and annotationslist to view the fixed menus.

### ADDITIONAL INFORMATION

- [ x] Has associated issue: https://github.com/apache/superset/issues/14617, https://github.com/apache/superset/issues/14616
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Fixes #14616, Fixes #14617

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 1]
```diff
@@ -111,8 +113,8 @@ const StyledHeader = styled.div`
   @media (max-width: 767px) {
     .header,
     .nav-right {
-      float: left;
-      padding-left: ${({ theme }) => theme.gridUnit * 2}px;
+      position: relative;
+      margin-left: 8px;
```
Comment (rusackas) [inline superset-frontend/src/components/Menu/SubMenu.tsx:117]:
```suggestion
      margin-left: ${({ theme }) => theme.gridUnit * 2}px;
```

**Comment by rusackas** [comment]
@pkdotson A similar (related?) issue was pointed out to me and I opened this quick PR. Hopefully I'm not going to cause a collision there...
https://github.com/apache/superset/pull/14637

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 2]
```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
...
-                      })}
                       role="tab"
+                      data-test={tab['data-test']}
+                      className={tab.name === props.activeChild ? 'active' : ''}
                     >
-                      <a href={tab.url} onClick={tab.onClick}>
-                        {tab.label}
-                      </a>
+                      <div>
+                        <Link to={tab.url || ''}>{tab.label}</Link>
```
Comment (rusackas) [inline superset-frontend/src/components/Menu/SubMenu.tsx:211]: Do all instances of SubMenu use react router, or do some of them require links? Just want to make sure switching to <Link> isn't going to break any functionality.

**Comment by michael-s-molina** [comment]
> Fixes #14616, Fixes #14617

GitHub will not automatically close these issues because it doesn't check for comments. @pkdotson should add this text to the PR description.

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 3]
```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
...
-                        </div>
-                      </li>
-                    </Menu.Item>
-                  );
-                }
-
+        {props.name && <div className="header">{props.name}</div>}
+        <Menu mode={showMenu} style={{ backgroundColor: 'transparent' }}>
+          {props.tabs &&
+            props.tabs.map(tab => {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Menu/SubMenu.tsx:201]:
```suggestion
          {props.tabs?.map(tab => {
```

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 4]
```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
...
-                        active: tab.name === props.activeChild,
-                      })}
                       role="tab"
+                      data-test={tab['data-test']}
+                      className={tab.name === props.activeChild ? 'active' : ''}
                     >
-                      <a href={tab.url} onClick={tab.onClick}>
-                        {tab.label}
-                      </a>
+                      <div>
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Menu/SubMenu.tsx:210]: Is this `div` required?

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 5]
```diff
@@ -42,6 +42,8 @@ const StyledHeader = styled.div`
     padding: 14px 0;
     margin-right: ${({ theme }) => theme.gridUnit * 3}px;
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Menu/SubMenu.tsx:43]: You can replace all instances of ${({ theme }) => by moving this piece of code to the top like this https://github.com/apache/superset/pull/14448/files#diff-0856ab0ce511fac9256389d5ed3635f15d77ec907de602ce32d28f072de76753

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 6]
```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
...
+            props.tabs.map(tab => {
+              if ((props.usesRouter || hasHistory) && !!tab.usesRouter) {
                 return (
                   <Menu.Item key={tab.label}>
                     <li
-                      className={cx('no-router', {
-                        active: tab.name === props.activeChild,
-                      })}
                       role="tab"
+                      data-test={tab['data-test']}
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Menu/SubMenu.tsx:207]: Is this required? You already have a `role` defined. If you need more distinction you can use `aria-label `also.

File: superset-frontend/src/components/Menu/SubMenu.tsx [hunk 7]
```diff
@@ -195,69 +192,57 @@ const SubMenuComponent: React.FunctionComponent<SubMenuProps> = props => {
...
+              );
+            })}
+        </Menu>
+        <div className={navRightStyle}>
+          {props.buttons?.map((btn, i) => (
+            <Button
+              key={i}
+              buttonStyle={btn.buttonStyle}
+              onClick={btn.onClick}
+              data-test={btn['data-test']}
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Menu/SubMenu.tsx:240]: You can query by button `role` and `name` here. `data-test` is not required.

File: superset-frontend/src/components/Menu/SubMenu.tsx (hunk 4 above)
Comment (pkdotson) [reply superset-frontend/src/components/Menu/SubMenu.tsx:210]: So the diff looks weird this is original business logic. I did not add anything.

File: superset-frontend/src/components/Menu/SubMenu.tsx (hunk 7 above)
Comment (pkdotson) [reply superset-frontend/src/components/Menu/SubMenu.tsx:240]: I can look into this as a follow up ticket but this is original logic.

File: superset-frontend/src/components/Menu/SubMenu.tsx (hunk 2 above)
Comment (pkdotson) [reply superset-frontend/src/components/Menu/SubMenu.tsx:211]: No the check is still there checking if a link is router link or regular tag link.

File: superset-frontend/src/components/Menu/SubMenu.tsx (hunk 6 above)
Comment (pkdotson) [reply superset-frontend/src/components/Menu/SubMenu.tsx]: same as prev comments these this is original logic.

**Review by rusackas — APPROVED** [review]
LGTM!

**Comment by junlincc** [comment]
LGTM!! Thanks for the fix 😊

File: superset-frontend/src/components/Menu/SubMenu.tsx (hunk 5 above)
Comment (rusackas) [reply superset-frontend/src/components/Menu/SubMenu.tsx:43]: Nice tip!! We should sweep through and do this all over the place, on some rainy day (probably not now).

**Pull request merged**